File: sdk-sample/src/main/java/com/example/cxensesdk/MainAdapter.java
package com.example.cxensesdk;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.Arrays;
import java.util.List;

/**
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */

public class MainAdapter extends RecyclerView.Adapter<MainAdapter.ViewHolder> {
    private List<String> data;
    private ItemClickListener clickListener;

    public MainAdapter(String[] data, ItemClickListener clickListener) {
        this.data = Arrays.asList(data);
        this.clickListener = clickListener;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        String item = data.get(position);
        holder.title.setText(item);
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    interface ItemClickListener {
        void onItemClick(String item);
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        public TextView title;

        public ViewHolder(View itemView) {
            super(itemView);
            title = (TextView) itemView.findViewById(R.id.title);
            itemView.setOnClickListener(v -> {
                int position = getAdapterPosition();
                if (position != RecyclerView.NO_POSITION && clickListener != null) {
                    clickListener.onItemClick(data.get(position));
                }
            });
        }
    }
}

File: sdk-sample/src/main/java/com/example/cxensesdk/MainAdapter.kt
package com.example.cxensesdk

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import by.kirich1409.viewbindingdelegate.viewBinding
import com.example.cxensesdk.databinding.ListItemBinding

/**
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */

class MainAdapter(
    private val data: List<String>,
    private val clickListener: (String) -> Unit
) : RecyclerView.Adapter<MainAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
        ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.list_item, parent, false))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.binding.title.text = data[position]
    }

    override fun getItemCount(): Int = data.size

    inner class ViewHolder(containerView: View) : RecyclerView.ViewHolder(containerView) {
        val binding: ListItemBinding by viewBinding()

        init {
            binding.root.setOnClickListener {
                val position = adapterPosition
                if (position != RecyclerView.NO_POSITION) {
                    clickListener(data[position])
                }
            }
        }
    }
}
